ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartUploadPartHandler.java

```diff
@@ -104,10 +101,10 @@ void handle(RestRequest restRequest, RestResponseChannel restResponseChannel,
     String uploadId = RestUtils.getHeader(restRequest.getArgs(), UPLOAD_ID_QUERY_PARAM, true);
     restRequest.setArg(S3_CHUNK_UPLOAD, true);
     restRequest.setArg(SESSION, uploadId);
-    // TODO [S3] : set part ttl value
-    // restRequest.setArg(TTL, xxx);
-
     restRequest.setArg(SEND_FAILURE_REASON, Boolean.TRUE);
+    if (Utils.Infinite_Time == RestUtils.getTtlFromRequestHeader(restRequest.getArgs())) {
+      restRequest.setArg(TTL, frontendConfig.permanentNamedBlobInitialPutTtl);
```

Comment:
I am wondering if `frontendConfig.chunkUploadInitialChunkTtlSecs` or `frontendConfig.chunkUploadMaxChunkTtlSecs` may be more appropriate configs? These are the default and max TTL we allow for chunks in Ambry multipart uploads.

Also, may be we don't need to check `RestUtils.getTtlFromRequestHeader(restRequest.getArgs())` since S3 requests won't have the `x-ambry-ttl` http header?

Reply:
Agreed and fixed